smoke-test.sh

```diff
@@ -198,6 +205,17 @@ function doCommand() {
             $killCommand
             wait $!
             echo "    terminated with exit code $?" | tee -a "$logFile"
+        elif [ "$1" == "publish" ]; then
+            binlogPrefix=$(echo "$testingDir/${lang}_${proj}_${binlogOnlinePart}_${binlogHttpsPart}_" | sed 's/--/_/g ; s/ /_/g')
+            if [ "$projectOutput" == "true" ]; then
+                "${dotnetCmd}" publish --self-contained false /bl:"${binlogPrefix}publish-fx-dep.binlog" | tee -a "$logFile"
+                "${dotnetCmd}" publish --self-contained true -r $targetRid /bl:"${binlogPrefix}publish-self-contained-${targetRid}.binlog" | tee -a "$logFile"
+                "${dotnetCmd}" publish --self-contained true -r linux-x64 /bl:"${binlogPrefix}publish-self-contained-portable.binlog" | tee -a "$logFile"
+            else
+                "${dotnetCmd}" publish --self-contained false /bl:"${binlogPrefix}publish-fx-dep.binlog" >> "$logFile" 2>&1
+                "${dotnetCmd}" publish --self-contained true -r $targetRid /bl:"${binlogPrefix}publish-self-contained-${targetRid}.binlog" >> "$logFile" 2>&1
+                "${dotnetCmd}" publish --self-contained true -r linux-x64 /bl:"${binlogPrefix}publish-self-contained-portable.binlog" >> "$logFile" 2>&1
```

Comment:
Define a quick function locally?

```sh
runPublishScenarios() {
  "${dotnetCmd}" publish --self-contained false /bl:"${binlogPrefix}publish-fx-dep.binlog"
  "${dotnetCmd}" publish --self-contained true -r $targetRid /bl:"${binlogPrefix}publish-self-contained-${targetRid}.binlog"
  "${dotnetCmd}" publish --self-contained true -r linux-x64 /bl:"${binlogPrefix}publish-self-contained-portable.binlog"
}
if [ "$projectOutput" == "true" ]; then
  runPublishScenarios | tee -a "$logFile"
else
  runPublishScenarios >> "$logFile" 2>&1
fi
```

Reply:
Cool, I did not know about that.  Changed as suggested.